### PR TITLE
test: Add comprehensive unit tests for pkg/stream/tunnel.go

### PR DESCRIPTION
**Pull request by Dhiren-Mhatre**
**What type of PR is this?**
/kind test

**What this PR does / why we need it**:
This PR adds comprehensive unit tests for the tunnel package in pkg/stream/tunnel.go to improve code coverage and reliability. The test suite includes coverage for DefaultTunnel initialization, message writing operations, WebSocket control messages, and connection handling. This work is part of the larger initiative to enhance KubeEdge's test coverage across core directories.

The added tests help ensure the robustness of the tunnel package which is crucial for the streaming functionality in KubeEdge. The tests are designed to work with the existing implementation while providing thorough coverage of both success and edge cases.

**Which issue(s) this PR fixes**:
Part of #6101 ([lfx-mentorship] Enhance KubeEdge testing coverage)

**Special notes for your reviewer**:
The test implementation follows these key principles:
1. Works with existing tunnel.go implementation without modifications
2. Provides comprehensive coverage including edge cases
3. Uses proper mock WebSocket server for integration testing
4. Implements proper resource cleanup and error handling
5. Contributes to the goal of achieving >60% test coverage in core directories

**Does this PR introduce a user-facing change?**:
NONE
 
# Snapshot
image

### CODE REVIEW

File: pkg/stream/tunnel_test.go [hunk 1]
```diff
@@ -0,0 +1,212 @@
...
+	defer conn.Close()
+
+	tunnel := NewDefaultTunnel(conn)
+
+	testCases := []struct {
+		name        string
+		messageType int
+		data       []byte
+		deadline    time.Time
+		wantErr    bool
```
Comment (wbc6080) [inline pkg/stream/tunnel_test.go:132]: plz fix lint error and squash some commits

Comment (Dhiren-Mhatre) [reply pkg/stream/tunnel_test.go]: Sure, I have fixed lint error and squashed commits so that this pr has single commit.

**Comment by Dhiren-Mhatre** [comment]
Hi @wbc6080  I'm noticing that the  multiple checks have failed. Is this due to an issue on GitHub's side, or is there something specific I need to correct? Because it says service unavailable.

**Comment by wbc6080** [comment]
> Hi @wbc6080 I'm noticing that the multiple checks have failed. Is this due to an issue on GitHub's side, or is there something specific I need to correct? Because it says service unavailable.

It seems to be a problem with github action. You can close and reopen this PR to try CI again.

**Comment by Dhiren-Mhatre** [comment]
> > Hi @wbc6080 I'm noticing that the multiple checks have failed. Is this due to an issue on GitHub's side, or is there something specific I need to correct? Because it says service unavailable.
> 
> It seems to be a problem with github action. You can close and reopen this PR to try CI again.

@wbc6080 I have closed and reopened the PR as suggested. Waiting for the CI checks to run again.

**Comment by wbc6080** [comment]
/lgtm
/remove-lifecycle stale
cc @WillardHu

**Review by Shelley-BaoYue — COMMENTED** [review]
/approve
